File: compiler/seed-c/fixtures/loop-conditional-early-return.c
// Expected: exit 0; stdout: "13,2,0\n"; stderr: ""
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>

static int64_t first_at(int64_t limit) {
  int64_t index = 0;
  while (index < limit) {
    index += 1;
    if (index == 3) {
      return index + 10;
    }
  }
  return index;
}

int main(void) {
  const int64_t early = first_at(5);
  const int64_t exhausted = first_at(2);
  const int64_t empty = first_at(0);
  if (printf("%" PRId64 ",%" PRId64 ",%" PRId64 "\n", early,
             exhausted, empty) < 0) {
    return 1;
  }
  return 0;
}

File: compiler/seed-c/fixtures/loop-conditional-early-return.rs
// Expected: exit 0; stdout: "13,2,0\n"; stderr: ""
fn first_at(limit: i64) -> i64 {
    let mut index: i64 = 0;
    while index < limit {
        index += 1;
        if index == 3 {
            return index + 10;
        }
    }
    index
}

fn main() {
    let early = first_at(5);
    let exhausted = first_at(2);
    let empty = first_at(0);
    println!("{early},{exhausted},{empty}");
}
